chia/data_layer/data_store.py

```diff
@@ -81,6 +81,24 @@ async def create(cls, db_wrapper: DBWrapper) -> "DataStore":
                 )
                 """
             )
+            # TODO: Add ancestor -> hash relationship, this might involve temporarily
+            # deferring the foreign key enforcement due to the insertion order
+            # and the node table also enforcing a similar relationship in the
+            # other direction.
+            # FOREIGN KEY(ancestor) REFERENCES ancestors(ancestor)
+            await self.db.execute(
+                """
+                CREATE TABLE IF NOT EXISTS ancestors(
+                    hash TEXT NOT NULL REFERENCES node,
+                    ancestor TEXT,
+                    tree_id TEXT NOT NULL,
+                    generation INTEGER NOT NULL,
+                    PRIMARY KEY(hash, tree_id, generation),
+                    FOREIGN KEY(tree_id, generation) REFERENCES root(tree_id, generation),
+                    FOREIGN KEY(ancestor) REFERENCES node(hash)
```

Comment:
I think, basically, all fields of type `TEXT` should be `BLOB`. let's not store hashes in hex.

Reply:
Long since a plan, now not a priority to merge vs. functional changes.  https://github.com/Chia-Network/chia-blockchain/pull/9284  It will still get in at some point.  I added a note there to review other changes that have been made in the mean time so we get all the `TEXT` switched over.